### PR TITLE
Allow optional padding prop to scale axis

### PR DESCRIPTION
**Pull request by CharlieMartell**
This PR lets a user supply some optional scaling props in order to add padding to their axis. This solves a problem that charts with large data sets with small standard deviations causing them to be deceptive in their visualization.

### CODE REVIEW

File: examples/zoomable-chart/yarn.lock [hunk 1]
```diff
@@ -1 +0,0 @@
-../yarn.lock
```
Comment (btford) [inline examples/zoomable-chart/yarn.lock]: Why delete this?

File: tests/utils/scales-utils-tests.js [hunk 2]
```diff
@@ -304,3 +348,7 @@ test('scales-utils #_adjustCategoricalScale', t => {
   });
   t.end();
 });
+
+test('scales-utils #missingScaleProps_padDomain', t => {
```
Comment (Apercu) [inline tests/utils/scales-utils-tests.js]: add a TODO or delete?

File: examples/zoomable-chart/yarn.lock (hunk 1 above)
Comment (CharlieMartell) [inline examples/zoomable-chart/yarn.lock]: this was removed and readded automatically by running the script that removes references to unpm

**Comment by mcnuttandrew** [comment]
I don't understand this pr, why can't you just set the domain via xDomaub or yDomain?

**Comment by mcnuttandrew** [comment]
Couple of additional comments

-there isn't any example usage or docs?
- this API doesn't match the rest of the react-vis's API. I would suggest instead xPadding and yPaadding rather than an object of padding, as this can't be re-used for other variables (r or theta)

**Comment by CharlieMartell** [comment]
You could pass in a domain via xDomain or yDomain, however the domain is calculated in scales-utils getDomainByAttr and if you are using react-vis and want to specify your own domain you have to copy the way react-vis calculates domain in order to modify it according to dynamic data. Allowing the user to scale their data for visualization seems in line with react-vis goal no?

With respect to it not matching the API that makes sense, I will go ahead and adjust to better fit with documentation.

**Comment by mcnuttandrew** [comment]
K I think I get it now. Maybe this could be expanded to include left and right padding? E.g. XLeftPadding & XRightPadding for additional specificity

Also, the user should prolly get a warning if they are supplying a domain and a padding as that'll probably lead to confusion

**Pull request merged**